### ISSUE TITLE
New ideas welcomed

### ISSUE THREAD

**Issue by riya-17** [riya-17/Neural-Artistic-Style-Transfer#1]
Any new thought like transferring style to only objects of an image.

**Comment by IbrarYunus** [comment]
Hi, I'm currently downloading the .MAT file as I write. Here are some of the ideas that I am willing to work on these days: 

1. Add Contributors.md 
2. Can I make a GUI of it? Perhaps a Photoshop plugin.
3. If we can increase processing speed - then maybe convert a whole video to art style.

**Comment by riya-17** [comment]
Hey, I am glad you want to work on this project :). 

1. Contributors.md is a good idea. It will help new contributors to get comfortable with project.
2. May I get some ideas about GUI implementation. How you are planning to do it?
3. What are your plans to increase the speed?

**Comment by IbrarYunus** [comment]
Thanks for the speedy reply. :) 

1. For Contributors.md, shall I add it to my fork and create a pull request?
2. I would need to look into how Photoshop plugins are created, for a standalone UI for Windows/Linux, I could use Tkintr or something better.
3. I will try to put some optimizations here and there and time functions to see how performance is affected. I am thinking of adding GPU acceleration with tensorflow-gpu.

**Comment by riya-17** [comment]
@IbrarYunus 
1. Yaa Sure you can send me Pull Request.
2. Okay, As far I read about tkinter, I think it would be a better option. 
3. You can surely add tensorflow-gpu as it will increase the speed. I haven't added because tensorflow-gpu isn't supported by AMD. So, i didn't have an option.

**Comment by LTKills** [comment]
Hi! I have some ideas of code and overall simplification

1. I wonder if there's an easier (automatic) way of getting the image's dimentions so that height and width wouldn't need to be hardcoded.
2. Use sys.argv to get images' paths avoiding it to be hardcoded.
3. Use pipenv (preferred) or simply a requirements.txt to make installing requirements easier

Obs.: I believe the ideas from @IbrarYunus should be on separate issues, and so should mine (if you agree)

**Comment by riya-17** [comment]
Hello @LTKills , It's nice to get new ideas and your interest.

1. I don't know about any other easier(automatic) way, I found this easy to implement and understand.
2. Yaa you can use sys.argv But I found this more comfortable in compiling it again and again. I have even mentioned the other way in comments.
3. Please do that if you feel comfortable.

Yaa, why didn't you created other issue and then reported this. No problem I wil do that ;)

**Comment by LTKills** [comment]
Heyy so I created a bunch of issues. Feel free to remove the ones you think are not relevant, edit what you want changed or think is not as well described as it should be and add useful labels.

:)

**Comment by riya-17** [comment]
Ok. I will do it. Thanks :)

**Comment by spartan-vishan** [comment]
It is tedious to download the weights from a website every time the project is cloned. We can have a simple in-code solution using the tf.keras.applications.vgg19 module. I can implement this, if the issue is created.

**Comment by spartan-vishan** [comment]
Rather than hard coding the neural net layer creation, perhaps an iterative solution would be optimal.

**Comment by riya-17** [comment]
@spartan-vishan glad to know you wanna contribute to this project. You may introduce the issue by creating new issue regarding weights download feature and start working on it :)

**Comment by riya-17** [comment]
@spartan-vishan I have used a hard coded neural net because it is comparably easy and understandable, and i wanted to keep it simple. Converting it to an iterative solution is a good idea if you would like to implement it create an issue regarding conversion and feel free to contribute to it :)
